packages/adapter-cloudflare/index.js

```diff
@@ -63,6 +62,35 @@ export default function (options = {}) {
 	};
 }
 
+/**
+ * There's a limit of 100 rules. Take advantage of wildcards and generate as few file paths as possible
+ * https://developers.cloudflare.com/pages/platform/functions/routing/#create-a-_routesjson-file
+ *
+ * @param {import('@sveltejs/kit').Builder} builder
+ * @param {string} file_path
+ */
+function find_shallowest_exclude_pattern(builder, file_path) {
+	const exclude_file_path = `/${file_path}`;
+	const path_segments = file_path.split('/');
+	if (path_segments.length === 0) {
+		return file_path;
+	}
+	let exclude_pattern = exclude_file_path;
+	// Exclude patterns can conflict with sveltekit routes, so don't wildcard paths in that case.
```

Comment:
from a quick glance at this, I can't tell exactly what the logic is that's being used, but I would guess it's not correct. E.g. I think you would need to consider whether a route segment has a matcher and it doesn't look like you're checking for that. https://kit.svelte.dev/docs/advanced-routing#matching

Reply:
I see, sveltekit has a lot of matching patterns.  informative.  I'm starting to think maybe I should add some tests before I publish a PR for this feature.  I would like to create a pr with reference to other packages.  On top of that, I will - work on this pr again 🙏

Thank you for taking time out of your busy schedule to review my PR.  I will make this PR a draft for now.